bot.py

```diff
@@ -333,6 +337,32 @@ async def notify_if_voice_client_is_missing(self, message):
             return True
         return False
 
+    async def playlist(self, message, command_content):
+        """Play a playlist"""
+        logging.info("Fetching playlist for user %s", message.author)
+        playlist = pytube.Playlist(command_content)
+        added_videos = 0
+        failed_videos = 0
+        for video in playlist:
+            try:
+                media = pafy.new(video)
+            except KeyError as err:
+                logging.error(err)
+                failed_videos += 1
+                continue
+            self.media_queue.put((media, message))
+            added_videos += 1
+            if added_videos == 1:
+                self.next_in_queue(notify=False)
```

Comment:
This is the only place where notify is set and the behavior it is disabling is anyway done in the line just below this one. Maybe just remove notify and the send below this line?

Reply:
Why would we want to skip to the next song here?

Reply:
Well... `next_in_queue` is not `async`, and therefore uses `loop.create_task` to send the notification. In my testing the message sending isn't performed immediately, and in some cases not until the playlist fetching has completed (>30 sec. later).

Reply:
I'm leaning both ways here. I don't like the added complexity in `next_in_queue`. But then I don't like the delay in starting playng the first song in the playlist.

@michael-je @thorduragust can you comment on this?

Reply:
Hmm I don't like the added complexity in `next_in_queue()` either, but I _do_ prefer that over a long delay before starting the playback. A few ideas:

1. Is it not possible or is there a very good reason not to make `next_in_queue()` async?
2. Is there a way to make all the pafy fetches asyncronous, rather than sequential?
3. The delay might be acceptable if the user gets more feedback. An idea here would be to notify the user at the start that we are _fetching playlist data_. Then maybe adding little check boxes `[ ] [ ] [ ]` or something (one for each video) that get filled with x's and o's `[x] [o]` (using the message edit feature) after each fetch, depending on whether it succeedes or fails. 

The last point might be a good idea anyway until we fix all the video fetch failures. We might even want to add in the titles of songs that succeed or fail, at the moment it's really frustrating when more than half of the playlist songs are dropped.

Reply:
see #56

Reply:
I made `next_in_queue()` `async`. This causes the `after_callback` to use `self.loop.create_task` to call `next_in_queue()`.

Added progress in text form. Added reaction to request for playlist message.